File: develop/server/clientProject/client/src/main/java/com/home/client/scene/unit/GUnit.java
package com.home.client.scene.unit;

import com.home.commonBase.constlist.generate.UnitType;
import com.home.commonBase.scene.unit.UnitFightLogic;
import com.home.commonBase.scene.unit.UnitIdentityLogic;
import com.home.commonBase.scene.unit.UnitMoveLogic;
import com.home.commonClient.scene.base.GameUnit;
import com.home.commonClient.scene.unit.CharacterIdentityLogic;

/** 游戏单位 */
public class GUnit extends GameUnit
{
	@Override
	protected UnitIdentityLogic createIdentityLogic()
	{
		switch(getType())
		{
			case UnitType.Character:
			{
				return new CharacterIdentityLogic();
			}
			//case UnitType.Puppet:
			//{
			//	return new GPuppetIdentityLogic();
			//}
		}
		
		return super.createIdentityLogic();
	}
	
	@Override
	protected UnitMoveLogic createMoveLogic()
	{
		return new GUnitMoveLogic();
	}
	
	@Override
	protected UnitFightLogic createFightLogic()
	{
		return new GUnitFightLogic();
	}
}

File: develop/client/game/Assets/src/game/scene/unit/GUnit.cs
using System;
using ShineEngine;

/// <summary>
/// 游戏单位
/// </summary>
public class GUnit:Unit
{
	public GUnitHeadLogic gHead;

	public GUnit()
	{

	}

	protected override void registLogics()
	{
		base.registLogics();

		gHead=(GUnitHeadLogic)head;

	}

	protected override UnitIdentityLogic createIdentityLogic()
	{
		switch(type)
		{
			case UnitType.Character:
			{
				return new GCharacterIdentityLogic();
			}
		}

		return base.createIdentityLogic();
	}

	protected override UnitPosLogic createPosLogic()
	{
		return new GUnitPosLogic();
	}

	protected override UnitFightLogic createFightLogic()
	{
		return new GUnitFightLogic();
	}

	protected override UnitShowLogic createShowLogic()
	{
		if(_type==UnitType.Character)
		{
			return new GCharacterShowLogic();
		}

		return new GUnitShowLogic();
	}

	protected override UnitMoveLogic createMoveLogic()
	{
		return new GUnitMoveLogic();
	}

	protected override UnitAvatarLogic createAvatarLogic()
	{
		return new GUnitAvatarLogic();
	}

	protected override UnitHeadLogic createHeadLogic()
	{
		return new GUnitHeadLogic();
	}

	protected override CharacterControlLogic createCharacterControlLogic()
	{
		return new GCharacterControlLogic();
	}
}